src/common/snippets/src/lowered/pass/pass.cpp

```diff
@@ -51,6 +51,26 @@ void PassPipeline::register_positioned_passes(const std::vector<PositionedPassLo
         register_pass(pp.position, pp.pass);
 }
 
+PassPipeline PassPipeline::merge_pipelines(const PassPipeline& lhs, const PassPipeline& rhs) {
+    OPENVINO_ASSERT(*lhs.get_pass_config() == *rhs.get_pass_config(), "2 passes with different PassConfigs can't be merged.");
+    const auto& lhs_passes = lhs.get_passes();
+    std::unordered_map<ov::DiscreteTypeInfo, std::shared_ptr<lowered::pass::PassBase>> passes_map;
+    for (const auto& pass : lhs_passes) {
+        passes_map[pass->get_type_info()] = pass;
```

Comment:
Note if there are two passes with similar `type_info`, we will save only the last one.
I think we need to assert that the map has the same size as `lhs_passes`

Reply:
Alternatively, we can merge all the `lhs_passes` with the above algorithm, so all same-type passes will be merged into one. Not sure we need to allow for same-type passes inside one handler though.

Reply:
> Note if there are two passes with similar `type_info`, we will save only the last one. I think we need to assert that the map has the same size as `lhs_passes`

Agree, the assert was added